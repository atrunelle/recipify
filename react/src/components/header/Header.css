.header {
  text-align: center;
  min-height: 400px; }
  .header__logo {
    height: 80px; }
  .header__text {
    margin-bottom: auto; }
