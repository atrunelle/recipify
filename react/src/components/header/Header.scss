
.header {
	text-align: center;
	height: 400px;
	
	&__logo {
		height: 80px;
	}

	&__title {
		font-size: 1.5em;
		text-transform: uppercase;
	}

	&__text {
		margin-bottom: auto; 
	}
}