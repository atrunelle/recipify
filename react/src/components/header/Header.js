import React, { Component } from 'react';
import logo from '../../assets/images/logo.svg';
import './Header.css';
import Menu from '../menu/Menu';
import { AppBar } from '@material-ui/core';

class Header extends Component {
  render() {
    return (
    <AppBar 
      className="header" 
      position="static" 
      elevation={0}>
        <Menu></Menu>
        <img src={logo} className="header__logo" alt="logo" />
        <h1 className="header__title">Recipify</h1>
        <p className="header__text">Create your own recipes by selecting ingredient and reviewing nutritional values before saving!</p>
    </AppBar>
    );
  }
}

export default Header;
