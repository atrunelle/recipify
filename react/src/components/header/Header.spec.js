import React from 'react';
import { mount } from 'enzyme';
import createRouterContext from 'react-router-test-context';
import PropTypes from 'prop-types';
import Header from './Header';

describe('Component: Header', () => {
  const context = createRouterContext()
  const childContextTypes = {
    router: PropTypes.object,
  };
  
  it('should match snapshot', () => {
    const component = mount(
      <Header />,
      { context, childContextTypes },
  );
    expect(component.html()).toMatchSnapshot();
  });
});