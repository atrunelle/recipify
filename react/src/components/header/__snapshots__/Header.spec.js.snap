// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should match snapshot 1`] = `"<header class=\\"MuiPaper-root-9 MuiPaper-elevation0-11 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7 header\\"><div class=\\"MuiToolbar-root-36 MuiToolbar-gutters-37\\"><div class=\\"MuiGrid-container-38 MuiGrid-justify-xs-flex-end-56\\"><nav><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-145 MuiButton-root-129\\" role=\\"button\\" href=\\"/\\"><span class=\\"MuiButton-label-130\\">Create recipe</span><span class=\\"MuiTouchRipple-root-148\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-145 MuiButton-root-129\\" role=\\"button\\" href=\\"/recipes/list\\"><span class=\\"MuiButton-label-130\\">My recipes</span><span class=\\"MuiTouchRipple-root-148\\"></span></a></nav></div></div><img src=\\"logo.svg\\" class=\\"header__logo\\" alt=\\"logo\\"><h1 class=\\"MuiTypography-root-155 MuiTypography-display3-157 header__title\\">Recipify</h1><p class=\\"header__text\\">Create your own recipes by selecting ingredient and reviewing nutritional values before saving!</p></header>"`;
